### PR TITLE
Add `process.paging.faults` metric to semantic conventions

### LINKED ISSUE #2809: Add process page fault metrics
**What are you trying to achieve?**

I want to add process [page fault](https://en.wikipedia.org/wiki/Page_fault) metrics to the semantic conventions. Here are the specific metrics I have in mind, relative to how Linux reports them:

- The number of major ("hard") page faults that the process has made
  - This is reported in `/proc/[pid]/stat` Linux file, described as `majflt` field in the [man page](http://man.he.net/man5/proc)
- The number of minor ("soft") page faults that the process has made
  - This is reported in `/proc/[pid]/stat` Linux file, described as `minflt` field in the [man page](http://man.he.net/man5/proc)
- The number of major ("hard") page faults that the process's waited-for children have made
  - This is reported in `/proc/[pid]/stat` Linux file, described as `cmajflt` field in the [man page](http://man.he.net/man5/proc)
- The number of minor ("soft") page faults that the process's waited-for children have made
  - This is reported in `/proc/[pid]/stat` Linux file, described as `cminflt` field in the [man page](http://man.he.net/man5/proc)

I propose the following names for the metrics:

- `process.memory.page_faults` with the attribute `type` of either `hard` or `soft`
- `process.memory.child_page_faults` with the attribute `type` of either `hard` or `soft`

**Additional context**

I have read through the discussions on putting information in metric names vs. attributes (#2589 , #2726). I considered various options for the metric names and attributes, including putting the `type` in the metric name and putting the `child` into an attribute (couldn't find a good name for this attribute). Please let me know if you have any thoughts on this.

Related issue to add these metrics in `hostmetrics` receiver: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14084.

### PR DESCRIPTION
**Pull request by astencel-sumo**
Fixes #2809 

## Changes

Adds a new metric `process.paging.faults` to the semantic conventions.

### CODE REVIEW

**Comment by joaopgrassi** [comment]
I wonder if we should wait for the build-tools PR to be merged https://github.com/open-telemetry/build-tools/pull/79 so metrics semconv can finally start being auto-generated. If we allow more coming it's just more work on whoever is going to bring them into YAML files.

**Comment by astencel-sumo** [comment]
> I wonder if we should wait for the build-tools PR to be merged [open-telemetry/build-tools#79](https://github.com/open-telemetry/build-tools/pull/79) so metrics semconv can finally start being auto-generated. If we allow more coming it's just more work on whoever is going to bring them into YAML files.

This is valid point. If this PR is very close to merging (less than a week), I believe it's better to wait. If it will take longer to merge the PR, perhaps it's better to unblock the changes to semantic conventions. I'm happy to help with the updates to the YAML files in any case.

**Review by jsuereth — APPROVED** [review]

**Comment by astencel-sumo** [comment]
What needs to happen for this pull request to be merged? (Other than me updating the branch again)

**Comment by jsuereth** [comment]
I don't think this needs to wait on the build-tools PR. IF there are enough approvals of this change, then it can go in first.

File: specification/metrics/semantic_conventions/process-metrics.md [hunk 1]
```diff
@@ -31,17 +31,18 @@ metrics](runtime-environment-metrics.md).
...
+| `process.context_switches`      | Counter                                            | {count}   | Number of times the process has been context switched.                                                                              | `type` SHOULD be one of: `involuntary`, `voluntary`                                                                                                                                             |
+| `process.paging.faults`         | Counter                                            | {faults}  | Number of page faults the process has made.                                                                                         | `type` SHOULD be one of: `major` (for major, or hard, page faults), `minor` (for minor, or soft, page faults).                                                                                  |
```
Comment (reyang) [inline specification/metrics/semantic_conventions/process-metrics.md:45]: I think we need to have something similar to `process.cpu.time`: "`type`, if specified, SHOULD be ...", there are situations where the process only knows the total count of hard/soft page faults.

**Review by reyang — APPROVED** [review]
LGTM.

**Pull request merged**